src/poetry/core/packages/dependency_group.py

```diff
@@ -22,35 +25,42 @@ def __init__(
         self._mixed_dynamic = mixed_dynamic
         self._dependencies: list[Dependency] = []
         self._poetry_dependencies: list[Dependency] = []
+        self._included_dependency_groups: dict[NormalizedName, DependencyGroup] = {}
 
     @property
     def name(self) -> str:
         return self._name
 
     @property
     def dependencies(self) -> list[Dependency]:
+        included_group_dependencies = self._resolve_included_dependency_groups()
+
         if not self._dependencies:
             # legacy mode
-            return self._poetry_dependencies
+            return self._poetry_dependencies + included_group_dependencies
```

Comment:
Maybe, we should switch from multiple returns to one return in the end so that we just have to add `included_group_dependencies` in one place instead of four?

Reply:
I tried something ...

Reply:
Looks good. The same for `dependencies_for_locking`, please. 😃